### DISCUSSION TITLE
Content multilanguage

### DISCUSSION THREAD

**Discussion by KratorD** [zikula/core#5237] [Q&A]
Hello all!
I am using Zikula 3.0.4 and Content 5.2.0. 
I'm not 100% sure, but I think everything was going well until I added a language to the website (it was in English and then I added Spanish).
I think since then, when I try to generate a new page, I get the validation error: "This value is too short. It should have 1 character or more."
What causes it and how can I correct it?
(I tried to look for this message in the Content code, but I couldn't find it)

Thanks.

**Comment by Guite** [comment]
Hello @KratorD,

can you maybe post a screenshot of your language configuration and translation folder/files? Also maybe a screenshot of the validation constraint message.

**Reply by KratorD** [reply]
Sure! Here we go!
![imagen](https://github.com/zikula/core/assets/5812569/31d81ed4-40b3-4c00-9139-84b7ac534c32)

![imagen](https://github.com/zikula/core/assets/5812569/63b84a57-7d57-412c-a24a-08142432c17b)

![imagen](https://github.com/zikula/core/assets/5812569/231b9066-268a-4984-b86a-058983d084f7)

Thanks

**Reply by Guite** [reply]
Hm. Maybe try using the `Add subpage` function in the context menu of a page in the page tree view (at `/content/admin/pages/view?tpl=tree`) to see if that one works?

**Reply by Guite** [reply]
Also you can check `vendor/symfony/symfony/src/Symfony/Component/Validator/Constraints/LengthValidator.php`. There is something like this:

```php
        if (null !== $constraint->min && $length < $constraint->min) {
            $this->context->buildViolation($constraint->min == $constraint->max ? $constraint->exactMessage : $constraint->minMessage)
                ->setParameter('{{ value }}', $this->formatValue($stringValue))
                ->setParameter('{{ limit }}', $constraint->min)
                ->setInvalidValue($value)
                ->setPlural((int) $constraint->min)
                ->setCode(Length::TOO_SHORT_ERROR)
                ->addViolation();
        }
```
Either debug into this to see the content of `$value` or try `->setCode(Length::TOO_SHORT_ERROR . ' [' . $value . ']')` to get it into the output.

**Reply by Guite** [reply]
It seems this is related to the `slug` field of the `Page` entity:

`@Assert\Length(min="1", max="255")`

**Reply by KratorD** [reply]
> Hm. Maybe try using the `Add subpage` function in the context menu of a page in the page tree view (at `/content/admin/pages/view?tpl=tree`) to see if that one works?

Same result

**Reply by KratorD** [reply]
> It seems this is related to the `slug` field of the `Page` entity:
> 
> `@Assert\Length(min="1", max="255")`

It's quite likely, but only when you have two languages activated. When you only have 1, it seems to work OK. What could you do to resolve this situation?

**Reply by KratorD** [reply]
About `->setCode(Length::TOO_SHORT_ERROR . ' [' . $value . ']')` 
I have added this code, then I change .env.local to env=dev and debug=1. But the message is the same
![imagen](https://github.com/zikula/core/assets/5812569/c99c3f58-37c6-4ffa-97f3-1feb6b4943ae)

**Reply by KratorD** [reply]
I have reviewed the symfony console and I can see this:
![imagen](https://github.com/zikula/core/assets/5812569/71dd3e6a-137b-4e6a-877d-2cb65f2c6c45)

So, yes! The problem is the slug.

**Reply by Guite** [reply]
The issue here is probably related to the combination of `Translatable` and `Sluggable` (this often made problems in the past at least).
If you do not require translatable slugs you can try removing `@Gedmo\Translatable` from `slug` property in `Content/Entity/Base/AbstractPageEntity.php`.

**Reply by KratorD** [reply]
Hello Guite.
I have already removed the Gedmo Translatable from the slug, but I have a question. What do I need to do now for the change to take effect in the field?
Thank you

**Reply by Guite** [reply]
I guess after you removed the translatable attribute and cleared the cache it should not try to use a different slug for each language anymore and should therefore not fail due to a missing Spanish slug.

**Reply by KratorD** [reply]
Hello Guite. 
Deleting the attribute and clearing the cache was not enough. I don't know whether to launch a schema update to see if it takes the slug attribute change. What do you think?

**Reply by Guite** [reply]
Maybe you will need to disable some more custom behavior, e.g. by adding `return;` in https://github.com/zikula-modules/Content/blob/main/src/extensions/Zikula/ContentModule/Listener/EntityLifecycleListener.php#L58

**Reply by KratorD** [reply]
Nope, still not working adding the return.

**Reply by Guite** [reply]
I guess you need to debug `/vendor/gedmo/doctrine-extensions/lib/Gedmo/Sluggable/SluggableListener.php` to see if `generateSlug()` is actually called at all.

**Reply by KratorD** [reply]
![imagen](https://github.com/zikula/core/assets/5812569/33df3bb3-98e7-4902-adab-c7ee7f1e9945)
the echo is not shown :(
